controller/model/posture_check_model_os_test.go

```diff
@@ -0,0 +1,304 @@
+		osCheck.OperatingSystems[0].OsType = os2Type
+
+		result := osCheck.Evaluate(postureData)
+
+		req := require.New(t)
+		req.False(result)
+	})
+
+	t.Run("returns false for valid os and no matching version", func(t *testing.T) {
+		osCheck, postureData := newMatchingOsCheckAndData()
+		postureData.Os.Type = os2Type
+		postureData.Os.Version = ""
+
+		result := osCheck.Evaluate(postureData)
+
+		req := require.New(t)
+		req.False(result)
+	})
+
+	t.Run("returns true for valid os and no required version and no submitted version", func(t *testing.T) {
+		osCheck, postureData := newMatchingOsCheckAndData()
+		postureData.Os.Type = os3Type
+		postureData.Os.Version = ""
+
+		result := osCheck.Evaluate(postureData)
+
+		req := require.New(t)
+		req.True(result)
+	})
+
+	t.Run("returns true for valid os and no required version and submitted version", func(t *testing.T) {
+		osCheck, postureData := newMatchingOsCheckAndData()
+		postureData.Os.Type = os3Type
+		postureData.Os.Version = "1.2.3"
+
+		result := osCheck.Evaluate(postureData)
+
+		req := require.New(t)
+		req.True(result)
+	})
+
+	t.Run("returns true for lower exact version match and later and higher match", func(t *testing.T) {
+		osCheck, postureData := newMatchingOsCheckAndData()
+		postureData.Os.Type = os2Type
+		postureData.Os.Version = os2Version1
+
+		result := osCheck.Evaluate(postureData)
+
+		req := require.New(t)
+		req.True(result)
+	})
+
+	t.Run("returns true for higher exact version match and an earlier match", func(t *testing.T) {
+		osCheck, postureData := newMatchingOsCheckAndData()
+		postureData.Os.Type = os2Type
+		postureData.Os.Version = "7.8.9"
+
+		result := osCheck.Evaluate(postureData)
+
+		req := require.New(t)
+		req.True(result)
+	})
+
+	//poorly formatted posture data
+	t.Run("returns false for posture data with valid os and partial os major version match", func(t *testing.T) {
+		osCheck, postureData := newMatchingOsCheckAndData()
+		postureData.Os.Version = "10"
+
+		result := osCheck.Evaluate(postureData)
+
+		req := require.New(t)
+		req.False(result)
+	})
+
+	t.Run("returns false for posture data with valid os and partial os major and minor version match", func(t *testing.T) {
+		osCheck, postureData := newMatchingOsCheckAndData()
+		postureData.Os.Version = "10.5"
+
+		result := osCheck.Evaluate(postureData)
+
+		req := require.New(t)
+		req.False(result)
+	})
+
+	t.Run("returns false for posture data with valid os and partial os major version match with dangling period", func(t *testing.T) {
+		osCheck, postureData := newMatchingOsCheckAndData()
+		postureData.Os.Version = "10"
+
+		result := osCheck.Evaluate(postureData)
+
+		req := require.New(t)
+		req.False(result)
+	})
+
+	t.Run("returns false for posture data with valid os and partial os major and minor version match with dangling period", func(t *testing.T) {
+		osCheck, postureData := newMatchingOsCheckAndData()
+		postureData.Os.Version = "10.5."
```

Comment:
how about a `10..5` test too? and `.5.5` and `0.2.0` or `0.0.0` maybe?

Reply:
At that point, we are testing the edge cases of the semver library I am using. I'd rather leave that testing in their court. They will all fail.